src/engine/network/index.ts

```diff
@@ -484,16 +486,69 @@ const messageView = createCursorView(new ArrayBuffer(1000));
 
 export function createPeerIdIndexMessage(state: GameState, peerId: string) {
   const input: NetPipeData = [state, messageView];
-  writeUint8(messageView, NetworkMessage.AssignPeerIdIndex);
+  writeMessageType(messageView, NetworkMessage.AssignPeerIdIndex);
   serializePeerIdIndex(input, peerId);
   return sliceCursorView(messageView);
 }
 
+/* Player NetworkId Message */
+
+export async function serializePlayerNetworkId(input: NetPipeData, peerId: string) {
+  const [state, cv] = input;
+  const peerEid = state.network.peerIdToEntityId.get(peerId);
+  const peerIdIndex = state.network.peerIdToIndex.get(peerId);
+  console.log(`serializePlayerNetworkId`, peerId, peerEid, peerIdIndex);
+  if (peerEid === undefined || peerIdIndex === undefined) {
+    console.error(
+      `could not send NetworkMessage.AssignPlayerEntity, ${peerId} not set on peerIdToEntity/peerIdToIndex map`
+    );
+    return input;
+  }
+
+  const peerNid = Networked.networkId[peerEid];
+
+  writeString(cv, peerId);
+  writeUint32(cv, peerNid);
+  return input;
+}
+
+export function deserializePlayerNetworkId(input: NetPipeData) {
+  const [state, cv] = input;
+  // read
+  const peerId = readString(cv);
+  const peerNid = readUint32(cv);
+
+  // hack - for some reason this message arrives before the peer entity is created
+  // despite being sent after the entity creation message
```

Comment:
culprit could be `postMessage`. either way solution for this will be to pack everything into one big packet instead of multiple to guarantee order on reception

Reply:
Yup, seems like a good idea regardless.